README language reworked (#1)
milossramek/translation-scripts: weblate/README.md

# Weblate

> Scripts related to translation of the LibreOffice user interface, help and other files. LibreOffice uses for translation the [Weblate](https://weblate.org) tool.

Author: ```suggestion
Scripts related to translation of the LibreOffice user interface, "help", and other files LibreOffice translates using the [Weblate](https://weblate.org) tool.
```

```
Abbreviations are removed on import.

> `-e`: Automatically translate substrings found in the 'ui' component. Based heuristics substrings which appear as UI strings are detected. A new column in output is added, with these substrings translated. These mixed strings can serve as a reference or can be translated by any of the machine translation services. If a translation is ambiguous, all possibilities are shown.

Author: I don't understand "Based heuristics substrings."

> `-e`: Automatically translate substrings found in the 'ui' component. Based heuristics substrings which appear as UI strings are detected. A new column in output is added, with these substrings translated. These mixed strings can serve as a reference or can be translated by any of the machine translation services. If a translation is ambiguous, all possibilities are shown.

Author: ```suggestion
`-e`: Automatically translate substrings found in the 'ui' component. Heuristics-based substrings which appear as UI strings are detected. A new column in output is added, with these substrings translated. These mixed strings can serve as a reference or can be translated by any of the machine translation services. If a translation is ambiguous, all possibilities are shown.
```
Or?

`-x lang{,lang}` Extra language to add to the export as reference (no space after ,). An additional column (columns) will be added to the export for reference.

## Usage scenarios
### Find and correct typos
Weblate does not have a tool to find typos. The proposed procedure consists of several steps: export strings, find typos by a spell-checker and correct the typos using the `sed` editor.
1. Export all strings without accelerators

`potrans.py -p sk -a export >en_sk.csv`

> Open the `en_sk.csv` file in LibreOffice, delete columns 1 - 3 (File name, Key ID, English strings) and save as `sk.csv`

Author: This is "KeyID" on line 143?

#### 2. Correct typos

1. Find the file with a typo (`typoo` below), for example with the command
```
grep typoo `find libo_ui-master -name [^.]\*.po`
```

> (`[^.]` ensures the backup files are not searched). Then open the file in a text editor and fix the problem.

Author: Or?

> 1. Change all typos at once using the `sed` editor. For that, from the list of typos create a file, e.g.,  `fixerror.sed` with a sed command for each typo in the form

Author: The "For that" is a bit iffy.

Check the changes by means of the `potrans.py -p ui diff` command. If something went wrong revert the changes (`potrans.pu -p ui re`), modify the sed file and run `sed` again.

#### 4. Upload the modifications
Upload the modifications by `potrans.py -p ui up` to the Weblate server.

## Translate strings using Google translate
Translation services as Google translate make sense only for longer text strings composed of sentences. GUI translation should be done directly in Weblate.
### 1. Export the desired substrings
Use the `export` command (eventually with one of its switches). If necessary, sort the CSV file according to text length and delete rows short on text.
### 2. Use the translation service

> Using LibreOffice, copy some rows in the Source column and translate them using https://translate.google.com/. Copy the translation to the Target column.

Author: I thought this was done in Weblate, https://docs.weblate.org/en/latest/admin/machine.html?highlight=machine%20translation#machine-translation which the earlier change about machine-translation is about.

### 3. Check and modify the translation
### 4. Import the translation
Import the translated CSV file using `potrans.py -p help -c infile.csv im`. The Google translate service adds a large amount of spaces (around tags, operators, the $ and % characters which a part of variables). These are removed upon import. This removal is probably not perfect, but if it happens to merge words, the import process is aborted and an error message with instructions is displayed.

# dialogtrans.py

> The script renders LibreOffice dialogs (*.ui files) and stores them as a PNG image with a pair of renditions: an English original (with strings supplemented by a corresponding key_id) and translated version:

Author: And now "key_id"?

<img src="img/updatedialog.png" width="800px" height="auto">

## The procedure
1. Copy the `libreofficedevX.Y/share/config/soffice.cfg` directory to the working directory
1. Download the weblate ui project using `potrans.py -p ui down` to the working directory
1. Run the script by `dialogtrans.py  soffice.cfg/xxx > trans.csv` to render all UI files in the `soffice.cfg/xxx` subdirectory (you may specify a single UI file to render just that)
1. Check the PNG renderings in `soffice.cfg/xxx/ui`, modify translations in trans.csv
1. Import `trans.csv` with `potrans.py -p ui -c trans.csv im`
1. Upload modified translation with `potrans.py -p ui up`

## Created files
1. A CSV file with all translatable strings is written to standard output. It may be modified and imported by the `potrans.py` tool.
1. For each UI file its two versions are created, with the `-key.ui` and `-lang.ui` suffix. They may be opened by the `glade` tool.

> 1. For each UI file a png image with its rendition is created.

Author: ```suggestion
1. For each UI file a PNG image with its rendition is created.
```
